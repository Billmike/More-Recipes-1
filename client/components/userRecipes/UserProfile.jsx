import React from 'react';
import PropTypes from 'prop-types';
import isEmpty from 'lodash/isEmpty';
import ReactTooltip from 'react-tooltip';

import avatar from '../../public/images/avatar.png';
import ProfileDetails from './ProfileDetails';
import EditProfileForm from './EditProfileForm';
import config from '../../config';
import Spinner from '../common/Spinner';

const propTypes = {
  fetchUserProfile: PropTypes.func.isRequired,
  editUserProfile: PropTypes.func.isRequired,
  uploadUserImage: PropTypes.func.isRequired,
  currentProfileUserId: PropTypes.number.isRequired,
  authenticatedUserId: PropTypes.number,
  user: PropTypes.shape({
    isFetchingUserProfile: PropTypes.bool,
    imageUploading: PropTypes.bool.isRequired,
    userProfile: PropTypes.shape({
      fullName: PropTypes.string,
      userImage: PropTypes.string,
    }),
    userImageUrl: PropTypes.string,
  }).isRequired,
};

const defaultProps = {
  authenticatedUserId: null,
};

/**
 * Class representing user profile section
 *
 * @class UserProfile
 *
 * @extends {React.Component}
 */
class UserProfile extends React.Component {
  /**
   * Creates an instance of UserProfile.
   *
   * @memberof UserProfile
   */
  constructor() {
    super();
    this.state = {
      isEditing: false,
      userDetails: null,
    };
  }

  /**
   * Component did mount life cycle method
   *
   * @returns {any} Dispatch action to fetch user profile
   *
   * @memberof UserProfile
   */
  componentDidMount() {
    window.scrollTo(0, 0);
    $('.materialboxed').materialbox();

    const { fetchUserProfile, currentProfileUserId } = this.props;
    fetchUserProfile(currentProfileUserId);
  }

  /**
   * Component did update life cycle method
   *
   * @returns {any} Initialize materialize class
   * @memberof UserProfile
   */
  componentDidUpdate() {
    $('.materialboxed').materialbox();
  }

  /**
   * Function to switch to edit profile form
   *
   * @param {object} userDetails - The details of the user
   *
   * @returns {any} Set editing to true
   */
  handleStartEdit = (userDetails) => {
    this.setState({
      userDetails,
      isEditing: true,
    });
    ReactTooltip.hide();
  }

  /**
   * Function to handle edit profile input fields
   *
   * @param {object} event - Edit profile events
   *
   * @returns {any} Set input value state values
   */
  handleProfileChange = (event) => {
    const { target: { name, value } } = event;
    const { userDetails } = this.state;
    this.setState({
      userDetails: {
        ...userDetails,
        [name]: value,
      }
    });
  }

  /**
   * Function to handle canceling of profile edit event
   *
   * @returns {any} Cancels the edit event
   */
  handleCancel = () => {
    this.setState({
      isEditing: false,
    });
  }

  /**
   * Function to handle submitting profile changes
   *
   * @returns {any} Submits the changes made
   *
   * @memberof UserProfile
   */
  handleSubmitProfile = () => {
    const { userDetails } = this.state;
    const { editUserProfile } = this.props;

    editUserProfile(userDetails)
      .then(() => {
        const { user: { error } } = this.props;
        if (isEmpty(error)) {
          this.setState({
            isEditing: false,
          });
        }
      });
  }

  /**
   * Function to handle profile picture upload
   *
   * @param {event} event - Picture upload event
   *
   * @returns {any} Uploads picture
   *
   * @memberof UserProfile
   */
  handleUploadPhoto = (event) => {
    event.preventDefault();
    const { target: { files } } = event;
    const file = files[0];
    const { uploadUserImage, editUserProfile } = this.props;
    const formData = new FormData();
    formData.append('file', file);
    formData.append('upload_preset', config.UPLOAD_PRESET);
    formData.append('api_key', config.API_KEY);

    uploadUserImage(formData)
      .then(() => {
        const { user: { userImageUrl, error } } = this.props;
        const imageFile = {
          userImage: userImageUrl,
        };

        if (isEmpty(error)) {
          editUserProfile(imageFile);
        }
      });
  }
  /**
   * Render user profile
   *
   * @returns {object} User profile
   *
   * @memberof UserProfile
   */
  renderUserProfile() {
    const { isEditing } = this.state;
    const {
      user: { imageUploading, userProfile }, currentProfileUserId,
      authenticatedUserId
    } = this.props;
    const { fullName, userImage } = userProfile;
    return (
      <div>
        {isEmpty(userProfile)
          ? <div className="center-align" >
            <h5>User does not exist</h5>
            <i className="material-icons large">folder_open</i>
          </div>
          : <div>
            <div className="user-image col s12 m5 l4">
              <div className="row">
                <figure className="col s12">
                  <img
                    id="profile-image"
                    className="materialboxed responsive-img circle"
                    src={userImage || avatar}
                    alt={fullName}
                  />
                </figure>
              </div>
              <span className="uploading-image-spinner">
                <div className="center-align">
                  {imageUploading && <Spinner size="small" />}
                </div>
              </span>
              <div className="row">
                {currentProfileUserId === authenticatedUserId
                  && <div className=" image-upload-button center">
                    <label
                      htmlFor="user-image"
                      className={`btn-floating waves-effect waves-light
                  ${userImage ? 'green' : 'indigo accent-2'}`}
                    >
                      <i className="material-icons small" data-tip="Edit Photo">
                        camera_alt
                      </i>
                    </label>
                    <input
                      id="user-image"
                      name="user-image"
                      type="file"
                      onChange={this.handleUploadPhoto}
                    />
                    <ReactTooltip />
                  </div>}
              </div>
            </div>
            <div className="col s12 m5 l6 z-depth-1">
              {isEditing && (currentProfileUserId === authenticatedUserId) ?
                <EditProfileForm
                  handleProfileChange={this.handleProfileChange}
                  handleSubmitProfile={this.handleSubmitProfile}
                  handleCancel={this.handleCancel}
                  {...this.state}
                  {...this.props}
                /> :
                <ProfileDetails
                  handleStartEdit={this.handleStartEdit}
                  {...this.props}
                />
              }
            </div>
          </div>
        }
      </div>
    );
  }

  /**
   * Render function
   *
   * @returns {object} React element
   *
   * @memberof UserProfile
   */
  render() {
    const { user: { isFetchingUserProfile } } = this.props;

    return (
      <div className="row">
        {isFetchingUserProfile
          ? <div className="center-align">
            <Spinner size="big" />
          </div>
          : this.renderUserProfile(this.props)
        }
      </div>
    );
  }
}

UserProfile.propTypes = propTypes;

UserProfile.defaultProps = defaultProps;

export default UserProfile;
