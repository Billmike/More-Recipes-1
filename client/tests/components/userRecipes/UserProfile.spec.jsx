import React from 'react';

import UserProfile from '../../../components/userRecipes/UserProfile';

describe('UserProfile component', () => {
  let spy;

  afterEach(() => {
    spy.mockClear();
  });
  it('should render component', () => {
    spy = jest.spyOn(UserProfile.prototype, 'componentDidMount');
    const props = {
      fetchUserProfile: jest.fn(),
      editUserProfile: jest.fn(),
      uploadUserImage: jest.fn(),
      currentProfileUserId: 1,
      authenticatedUserId: 1,
      user: {
        isFetchingUserProfile: false,
        imageUploading: false,
        userProfile: {
          fullName: 'John Mark',
          userImage: 'UseR image URL',
        },
        userImageUrl: '',
      },
    };
    const wrapper = shallow(<UserProfile {...props} />);

    expect(wrapper).toBeDefined();
    expect(spy).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
  it('should show message for non existing user', () => {
    const props = {
      fetchUserProfile: jest.fn(),
      editUserProfile: jest.fn(),
      uploadUserImage: jest.fn(),
      currentProfileUserId: 567,
      authenticatedUserId: 1,
      user: {
        isFetchingUserProfile: false,
        imageUploading: false,
        userProfile: {},
        userImageUrl: '',
      },
    };
    const wrapper = shallow(<UserProfile {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('h5').text()).toEqual('User does not exist');
    expect(wrapper).toMatchSnapshot();
  });
  it('should show spinner when loading user profile', () => {
    const props = {
      fetchUserProfile: jest.fn(),
      editUserProfile: jest.fn(),
      uploadUserImage: jest.fn(),
      currentProfileUserId: 1,
      authenticatedUserId: 1,
      user: {
        isFetchingUserProfile: true,
        imageUploading: false,
        userProfile: {
          fullName: 'John Mark',
          userImage: 'User image URL',
        },
        userImageUrl: '',
      },
    };
    const wrapper = shallow(<UserProfile {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Spinner').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should show spinner when uploading image', () => {
    const props = {
      fetchUserProfile: jest.fn(),
      editUserProfile: jest.fn(),
      uploadUserImage: jest.fn(),
      currentProfileUserId: 1,
      authenticatedUserId: 1,
      user: {
        isLoading: false,
        imageUploading: true,
        userProfile: {
          fullName: 'John Mark',
          userImage: null,
        },
        userImageUrl: '',
      },
    };
    const wrapper = shallow(<UserProfile {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Spinner').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should simulate click event', () => {
    const props = {
      fetchUserProfile: jest.fn(),
      editUserProfile: jest.fn(),
      uploadUserImage: jest.fn(() => Promise.resolve()),
      currentProfileUserId: 1,
      authenticatedUserId: 1,
      user: {
        isLoading: false,
        imageUploading: false,
        userProfile: {
          fullName: 'John Mark',
          userImage: 'User image URL',
        },
        userImageUrl: 'User new image URL',
        error: {}
      },
    };
    const wrapper = mount(<UserProfile {...props} />);

    expect(wrapper).toBeDefined();
    wrapper.find('input#user-image').simulate('change', {
      target: { files: [{ imageUrl: 'Use new image' }] },
      preventDefault: jest.fn(),
    });
    expect(wrapper).toMatchSnapshot();
  });
  it('should simulate edit profile click event', () => {
    const props = {
      fetchUserProfile: jest.fn(),
      editUserProfile: jest.fn(() => Promise.resolve()),
      uploadUserImage: jest.fn(() => Promise.resolve()),
      currentProfileUserId: 1,
      authenticatedUserId: 1,
      user: {
        isLoading: false,
        imageUploading: false,
        userProfile: {
          fullName: 'John Mark',
          userImage: 'User image URL',
        },
        userImageUrl: 'User new image URL',
        error: {}
      },
    };
    const wrapper = mount(<UserProfile {...props} />);
    spy = jest.spyOn(wrapper.instance(), 'handleStartEdit');
    wrapper.instance().forceUpdate();

    expect(wrapper).toBeDefined();
    wrapper.find('button#edit-btn').simulate('click');
    expect(wrapper.state('isEditing')).toEqual(true);
    wrapper.find('input[name="firstName"]').simulate('change', {
      target: { name: 'firstName', value: 'JOHN' },
    });
    expect(spy).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
  it('should simulate submit profile action', () => {
    const props = {
      fetchUserProfile: jest.fn(),
      editUserProfile: jest.fn(() => Promise.resolve()),
      uploadUserImage: jest.fn(() => Promise.resolve()),
      currentProfileUserId: 1,
      authenticatedUserId: 1,
      user: {
        isLoading: false,
        imageUploading: false,
        userProfile: {
          fullName: 'John Mark',
          userImage: 'User image URL',
        },
        userImageUrl: 'User new image URL',
        error: {}
      },
    };
    const wrapper = mount(<UserProfile {...props} />);
    spy = jest.spyOn(wrapper.instance(), 'handleSubmitProfile');
    wrapper.instance().forceUpdate();

    expect(wrapper).toBeDefined();
    wrapper.find('button#edit-btn').simulate('click');
    wrapper.find('button#confirm-edit-btn').simulate('click');
    expect(spy).toBeCalled();
  });
  it('should simulate close edit profile action', () => {
    const props = {
      fetchUserProfile: jest.fn(),
      editUserProfile: jest.fn(),
      uploadUserImage: jest.fn(() => Promise.resolve()),
      currentProfileUserId: 1,
      authenticatedUserId: 1,
      user: {
        isLoading: false,
        imageUploading: false,
        userProfile: {
          fullName: 'John Mark',
          userImage: 'User image URL',
        },
        userImageUrl: 'User new image URL',
        error: {}
      },
    };
    const wrapper = mount(<UserProfile {...props} />);
    spy = jest.spyOn(wrapper.instance(), 'handleCancel');
    wrapper.instance().forceUpdate();

    expect(wrapper).toBeDefined();
    wrapper.find('button#edit-btn').simulate('click');
    expect(wrapper.state('isEditing')).toEqual(true);
    wrapper.find('button#cancel-edit-btn').simulate('click');
    expect(spy).toBeCalled();
    expect(wrapper.state('isEditing')).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });
});
