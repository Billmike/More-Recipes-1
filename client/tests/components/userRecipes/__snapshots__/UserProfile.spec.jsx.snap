// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile component should render component 1`] = `
<div
  className="row"
>
  <div>
    <div>
      <div
        className="user-image col s12 m5 l4"
      >
        <div
          className="row"
        >
          <figure
            className="col s12"
          >
            <img
              alt="John Mark"
              className="materialboxed responsive-img circle"
              id="profile-image"
              src="UseR image URL"
            />
          </figure>
        </div>
        <span
          className="uploading-image-spinner"
        >
          <div
            className="center-align"
          />
        </span>
        <div
          className="row"
        >
          <div
            className=" image-upload-button center"
          >
            <label
              className="btn-floating waves-effect waves-light
                  green"
              htmlFor="user-image"
            >
              <i
                className="material-icons small"
                data-tip="Edit Photo"
              >
                camera_alt
              </i>
            </label>
            <input
              id="user-image"
              name="user-image"
              onChange={[Function]}
              type="file"
            />
            <ReactTooltip
              insecure={true}
              resizeHide={true}
              wrapper="div"
            />
          </div>
        </div>
      </div>
      <div
        className="col s12 m5 l6 z-depth-1"
      >
        <ProfileDetails
          authenticatedUserId={1}
          currentProfileUserId={1}
          editUserProfile={[MockFunction]}
          fetchUserProfile={
            [MockFunction] {
              "calls": Array [
                Array [
                  1,
                ],
              ],
            }
          }
          handleStartEdit={[Function]}
          uploadUserImage={[MockFunction]}
          user={
            Object {
              "imageUploading": false,
              "isFetchingUserProfile": false,
              "userImageUrl": "",
              "userProfile": Object {
                "fullName": "John Mark",
                "userImage": "UseR image URL",
              },
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`UserProfile component should show message for non existing user 1`] = `
<div
  className="row"
>
  <div>
    <div
      className="center-align"
    >
      <h5>
        User does not exist
      </h5>
      <i
        className="material-icons large"
      >
        folder_open
      </i>
    </div>
  </div>
</div>
`;

exports[`UserProfile component should show spinner when loading user profile 1`] = `
<div
  className="row"
>
  <div
    className="center-align"
  >
    <Spinner
      size="big"
    />
  </div>
</div>
`;

exports[`UserProfile component should show spinner when uploading image 1`] = `
<div
  className="row"
>
  <div>
    <div>
      <div
        className="user-image col s12 m5 l4"
      >
        <div
          className="row"
        >
          <figure
            className="col s12"
          >
            <img
              alt="John Mark"
              className="materialboxed responsive-img circle"
              id="profile-image"
              src="test-file-stub"
            />
          </figure>
        </div>
        <span
          className="uploading-image-spinner"
        >
          <div
            className="center-align"
          >
            <Spinner
              size="small"
            />
          </div>
        </span>
        <div
          className="row"
        >
          <div
            className=" image-upload-button center"
          >
            <label
              className="btn-floating waves-effect waves-light
                  indigo accent-2"
              htmlFor="user-image"
            >
              <i
                className="material-icons small"
                data-tip="Edit Photo"
              >
                camera_alt
              </i>
            </label>
            <input
              id="user-image"
              name="user-image"
              onChange={[Function]}
              type="file"
            />
            <ReactTooltip
              insecure={true}
              resizeHide={true}
              wrapper="div"
            />
          </div>
        </div>
      </div>
      <div
        className="col s12 m5 l6 z-depth-1"
      >
        <ProfileDetails
          authenticatedUserId={1}
          currentProfileUserId={1}
          editUserProfile={[MockFunction]}
          fetchUserProfile={
            [MockFunction] {
              "calls": Array [
                Array [
                  1,
                ],
              ],
            }
          }
          handleStartEdit={[Function]}
          uploadUserImage={[MockFunction]}
          user={
            Object {
              "imageUploading": true,
              "isLoading": false,
              "userImageUrl": "",
              "userProfile": Object {
                "fullName": "John Mark",
                "userImage": null,
              },
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`UserProfile component should simulate click event 1`] = `
<UserProfile
  authenticatedUserId={1}
  currentProfileUserId={1}
  editUserProfile={[MockFunction]}
  fetchUserProfile={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  uploadUserImage={
    [MockFunction] {
      "calls": Array [
        Array [
          FormData {
            Symbol(impl): FormDataImpl {
              "_entries": Array [
                Object {
                  "name": "file",
                  "value": "[object Object]",
                },
                Object {
                  "name": "upload_preset",
                  "value": "undefined",
                },
                Object {
                  "name": "api_key",
                  "value": "undefined",
                },
              ],
              Symbol(wrapper): [Circular],
            },
          },
        ],
      ],
    }
  }
  user={
    Object {
      "error": Object {},
      "imageUploading": false,
      "isLoading": false,
      "userImageUrl": "User new image URL",
      "userProfile": Object {
        "fullName": "John Mark",
        "userImage": "User image URL",
      },
    }
  }
>
  <div
    className="row"
  >
    <div>
      <div>
        <div
          className="user-image col s12 m5 l4"
        >
          <div
            className="row"
          >
            <figure
              className="col s12"
            >
              <img
                alt="John Mark"
                className="materialboxed responsive-img circle"
                id="profile-image"
                src="User image URL"
              />
            </figure>
          </div>
          <span
            className="uploading-image-spinner"
          >
            <div
              className="center-align"
            />
          </span>
          <div
            className="row"
          >
            <div
              className=" image-upload-button center"
            >
              <label
                className="btn-floating waves-effect waves-light
                  green"
                htmlFor="user-image"
              >
                <i
                  className="material-icons small"
                  data-tip="Edit Photo"
                >
                  camera_alt
                </i>
              </label>
              <input
                id="user-image"
                name="user-image"
                onChange={[Function]}
                type="file"
              />
              <ReactTooltip
                insecure={true}
                resizeHide={true}
                wrapper="div"
              >
                <div
                  className="__react_component_tooltip place-top type-dark "
                  data-id="tooltip"
                />
              </ReactTooltip>
            </div>
          </div>
        </div>
        <div
          className="col s12 m5 l6 z-depth-1"
        >
          <ProfileDetails
            authenticatedUserId={1}
            currentProfileUserId={1}
            editUserProfile={[MockFunction]}
            fetchUserProfile={
              [MockFunction] {
                "calls": Array [
                  Array [
                    1,
                  ],
                ],
              }
            }
            handleStartEdit={[Function]}
            uploadUserImage={
              [MockFunction] {
                "calls": Array [
                  Array [
                    FormData {
                      Symbol(impl): FormDataImpl {
                        "_entries": Array [
                          Object {
                            "name": "file",
                            "value": "[object Object]",
                          },
                          Object {
                            "name": "upload_preset",
                            "value": "undefined",
                          },
                          Object {
                            "name": "api_key",
                            "value": "undefined",
                          },
                        ],
                        Symbol(wrapper): [Circular],
                      },
                    },
                  ],
                ],
              }
            }
            user={
              Object {
                "error": Object {},
                "imageUploading": false,
                "isLoading": false,
                "userImageUrl": "User new image URL",
                "userProfile": Object {
                  "fullName": "John Mark",
                  "userImage": "User image URL",
                },
              }
            }
          >
            <div>
              <div
                className="row"
              />
              <div
                className="col s12"
              >
                <h5>
                  <i
                    className="material-icons left"
                  >
                    person
                  </i>
                  My
                   Profile
                </h5>
                <div
                  className="divider black"
                />
                <br />
                <div
                  className="row"
                  id="about_user"
                >
                  <div
                    className="row"
                    id="profile_username"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="username"
                    >
                      Username:
                    </label>
                    <div
                      className="col s9"
                      id="username"
                    />
                  </div>
                  <div
                    className="row"
                    id="profile_first_name"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="firstname"
                    >
                      First Name:
                    </label>
                    <div
                      className="col s9"
                      id="firstname"
                    />
                  </div>
                  <div
                    className="row"
                    id="profile_last_name"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="lastname"
                    >
                      Last Name:
                    </label>
                    <div
                      className="col s9"
                      id="lastname"
                    />
                  </div>
                  <div
                    className="row"
                    id="profile_email"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="email"
                    >
                      Email:
                    </label>
                    <div
                      className="col s9"
                      id="email"
                    />
                  </div>
                </div>
              </div>
              <div
                className="col s12"
              >
                <h5>
                  About 
                  Me
                </h5>
                <div
                  className="divider black"
                />
                <p
                  id="about-me"
                />
              </div>
              <div>
                <button
                  className="btn-floating waves-effect waves-light green right"
                  id="edit-btn"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons left"
                    data-tip="Edit Details"
                  >
                    edit
                  </i>
                </button>
                <ReactTooltip
                  insecure={true}
                  resizeHide={true}
                  wrapper="div"
                >
                  <div
                    className="__react_component_tooltip place-top type-dark "
                    data-id="tooltip"
                  />
                </ReactTooltip>
              </div>
              <div
                className="row"
              />
            </div>
          </ProfileDetails>
        </div>
      </div>
    </div>
  </div>
</UserProfile>
`;

exports[`UserProfile component should simulate close edit profile action 1`] = `
<UserProfile
  authenticatedUserId={1}
  currentProfileUserId={1}
  editUserProfile={[MockFunction]}
  fetchUserProfile={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  uploadUserImage={[MockFunction]}
  user={
    Object {
      "error": Object {},
      "imageUploading": false,
      "isLoading": false,
      "userImageUrl": "User new image URL",
      "userProfile": Object {
        "fullName": "John Mark",
        "userImage": "User image URL",
      },
    }
  }
>
  <div
    className="row"
  >
    <div>
      <div>
        <div
          className="user-image col s12 m5 l4"
        >
          <div
            className="row"
          >
            <figure
              className="col s12"
            >
              <img
                alt="John Mark"
                className="materialboxed responsive-img circle"
                id="profile-image"
                src="User image URL"
              />
            </figure>
          </div>
          <span
            className="uploading-image-spinner"
          >
            <div
              className="center-align"
            />
          </span>
          <div
            className="row"
          >
            <div
              className=" image-upload-button center"
            >
              <label
                className="btn-floating waves-effect waves-light
                  green"
                htmlFor="user-image"
              >
                <i
                  className="material-icons small"
                  data-tip="Edit Photo"
                >
                  camera_alt
                </i>
              </label>
              <input
                id="user-image"
                name="user-image"
                onChange={[Function]}
                type="file"
              />
              <ReactTooltip
                insecure={true}
                resizeHide={true}
                wrapper="div"
              >
                <div
                  className="__react_component_tooltip place-top type-dark "
                  data-id="tooltip"
                />
              </ReactTooltip>
            </div>
          </div>
        </div>
        <div
          className="col s12 m5 l6 z-depth-1"
        >
          <ProfileDetails
            authenticatedUserId={1}
            currentProfileUserId={1}
            editUserProfile={[MockFunction]}
            fetchUserProfile={
              [MockFunction] {
                "calls": Array [
                  Array [
                    1,
                  ],
                ],
              }
            }
            handleStartEdit={[Function]}
            uploadUserImage={[MockFunction]}
            user={
              Object {
                "error": Object {},
                "imageUploading": false,
                "isLoading": false,
                "userImageUrl": "User new image URL",
                "userProfile": Object {
                  "fullName": "John Mark",
                  "userImage": "User image URL",
                },
              }
            }
          >
            <div>
              <div
                className="row"
              />
              <div
                className="col s12"
              >
                <h5>
                  <i
                    className="material-icons left"
                  >
                    person
                  </i>
                  My
                   Profile
                </h5>
                <div
                  className="divider black"
                />
                <br />
                <div
                  className="row"
                  id="about_user"
                >
                  <div
                    className="row"
                    id="profile_username"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="username"
                    >
                      Username:
                    </label>
                    <div
                      className="col s9"
                      id="username"
                    />
                  </div>
                  <div
                    className="row"
                    id="profile_first_name"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="firstname"
                    >
                      First Name:
                    </label>
                    <div
                      className="col s9"
                      id="firstname"
                    />
                  </div>
                  <div
                    className="row"
                    id="profile_last_name"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="lastname"
                    >
                      Last Name:
                    </label>
                    <div
                      className="col s9"
                      id="lastname"
                    />
                  </div>
                  <div
                    className="row"
                    id="profile_email"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="email"
                    >
                      Email:
                    </label>
                    <div
                      className="col s9"
                      id="email"
                    />
                  </div>
                </div>
              </div>
              <div
                className="col s12"
              >
                <h5>
                  About 
                  Me
                </h5>
                <div
                  className="divider black"
                />
                <p
                  id="about-me"
                />
              </div>
              <div>
                <button
                  className="btn-floating waves-effect waves-light green right"
                  id="edit-btn"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons left"
                    data-tip="Edit Details"
                  >
                    edit
                  </i>
                </button>
                <ReactTooltip
                  insecure={true}
                  resizeHide={true}
                  wrapper="div"
                >
                  <div
                    className="__react_component_tooltip place-top type-dark "
                    data-id="tooltip"
                  />
                </ReactTooltip>
              </div>
              <div
                className="row"
              />
            </div>
          </ProfileDetails>
        </div>
      </div>
    </div>
  </div>
</UserProfile>
`;

exports[`UserProfile component should simulate edit profile click event 1`] = `
<UserProfile
  authenticatedUserId={1}
  currentProfileUserId={1}
  editUserProfile={[MockFunction]}
  fetchUserProfile={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  uploadUserImage={[MockFunction]}
  user={
    Object {
      "error": Object {},
      "imageUploading": false,
      "isLoading": false,
      "userImageUrl": "User new image URL",
      "userProfile": Object {
        "fullName": "John Mark",
        "userImage": "User image URL",
      },
    }
  }
>
  <div
    className="row"
  >
    <div>
      <div>
        <div
          className="user-image col s12 m5 l4"
        >
          <div
            className="row"
          >
            <figure
              className="col s12"
            >
              <img
                alt="John Mark"
                className="materialboxed responsive-img circle"
                id="profile-image"
                src="User image URL"
              />
            </figure>
          </div>
          <span
            className="uploading-image-spinner"
          >
            <div
              className="center-align"
            />
          </span>
          <div
            className="row"
          >
            <div
              className=" image-upload-button center"
            >
              <label
                className="btn-floating waves-effect waves-light
                  green"
                htmlFor="user-image"
              >
                <i
                  className="material-icons small"
                  data-tip="Edit Photo"
                >
                  camera_alt
                </i>
              </label>
              <input
                id="user-image"
                name="user-image"
                onChange={[Function]}
                type="file"
              />
              <ReactTooltip
                insecure={true}
                resizeHide={true}
                wrapper="div"
              >
                <div
                  className="__react_component_tooltip place-top type-dark "
                  data-id="tooltip"
                />
              </ReactTooltip>
            </div>
          </div>
        </div>
        <div
          className="col s12 m5 l6 z-depth-1"
        >
          <EditProfileForm
            authenticatedUserId={1}
            currentProfileUserId={1}
            editUserProfile={[MockFunction]}
            fetchUserProfile={
              [MockFunction] {
                "calls": Array [
                  Array [
                    1,
                  ],
                ],
              }
            }
            handleCancel={[Function]}
            handleProfileChange={[Function]}
            handleSubmitProfile={[Function]}
            isEditing={true}
            uploadUserImage={[MockFunction]}
            user={
              Object {
                "error": Object {},
                "imageUploading": false,
                "isLoading": false,
                "userImageUrl": "User new image URL",
                "userProfile": Object {
                  "fullName": "John Mark",
                  "userImage": "User image URL",
                },
              }
            }
            userDetails={
              Object {
                "aboutMe": undefined,
                "firstName": "JOHN",
                "lastName": undefined,
              }
            }
          >
            <div>
              <div
                className="row"
              />
              <h5>
                <i
                  className="material-icons left"
                >
                  person
                </i>
                My Profile
              </h5>
              <div
                className="divider black"
              />
              <br />
              <div
                className="row"
                id="about_user"
              >
                <TextField
                  className="validate"
                  disabled={true}
                  id="Username"
                  name="username"
                  onChange={[Function]}
                  placeholder="Username"
                  type="text"
                >
                  <div
                    className="row"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="username"
                    >
                      Username
                      :
                    </label>
                    <div
                      className="col s9"
                    >
                      <input
                        className="validate"
                        disabled={true}
                        name="username"
                        onChange={[Function]}
                        placeholder="Username"
                        type="text"
                      />
                    </div>
                  </div>
                </TextField>
                <TextField
                  className="validate"
                  id="First Name"
                  maxLength="255"
                  name="firstName"
                  onChange={[Function]}
                  placeholder="First name"
                  required={true}
                  type="text"
                >
                  <div
                    className="row"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="firstName"
                    >
                      First Name
                      :
                    </label>
                    <div
                      className="col s9"
                    >
                      <input
                        className="validate"
                        maxLength="255"
                        name="firstName"
                        onChange={[Function]}
                        placeholder="First name"
                        required={true}
                        type="text"
                      />
                    </div>
                  </div>
                </TextField>
                <TextField
                  className="validate"
                  id="Last Name"
                  maxLength="255"
                  name="lastName"
                  onChange={[Function]}
                  placeholder="Last name"
                  required={true}
                  type="text"
                >
                  <div
                    className="row"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="lastName"
                    >
                      Last Name
                      :
                    </label>
                    <div
                      className="col s9"
                    >
                      <input
                        className="validate"
                        maxLength="255"
                        name="lastName"
                        onChange={[Function]}
                        placeholder="Last name"
                        required={true}
                        type="text"
                      />
                    </div>
                  </div>
                </TextField>
                <TextField
                  disabled={true}
                  id="Email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="email"
                >
                  <div
                    className="row"
                  >
                    <label
                      className="col s3 black-text"
                      htmlFor="email"
                    >
                      Email
                      :
                    </label>
                    <div
                      className="col s9"
                    >
                      <input
                        disabled={true}
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="email"
                      />
                    </div>
                  </div>
                </TextField>
              </div>
              <div
                className="col s12"
              >
                <h5>
                  About Me
                </h5>
                <div
                  className="divider black"
                />
                <textarea
                  className="materialize-textarea"
                  id="aboutMe"
                  name="aboutMe"
                  onChange={[Function]}
                  placeholder="About me"
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="center-align"
                />
              </div>
              <div>
                <button
                  className="btn-floating waves-effect waves-light indigo accent-2
            right"
                  disabled={false}
                  id="confirm-edit-btn"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons"
                    data-tip="Save Details"
                  >
                    save
                  </i>
                </button>
                <button
                  className="btn-floating waves-effect waves-light red right"
                  disabled={false}
                  id="cancel-edit-btn"
                  onClick={[Function]}
                >
                  <i
                    className="material-icons"
                    data-tip="Cancel"
                  >
                    cancel
                  </i>
                </button>
                <ReactTooltip
                  insecure={true}
                  resizeHide={true}
                  wrapper="div"
                >
                  <div
                    className="__react_component_tooltip place-top type-dark "
                    data-id="tooltip"
                  />
                </ReactTooltip>
              </div>
              <div
                className="row"
              />
            </div>
          </EditProfileForm>
        </div>
      </div>
    </div>
  </div>
</UserProfile>
`;
